.github/workflows/build.yaml

```diff
@@ -0,0 +1,478 @@
+name: Build
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+
+jobs:
+  pre-build-updater:
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/stackrox-io/apollo-ci:scanner-test-0.3.61
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - name: Cache Go dependencies
+        uses: ./.github/actions/cache-go-dependencies
+
+      - name: Build updater
+        run: make build-updater
+
+      - name: Bundle the build to preserve permissions
+        run: tar -cvzf updater-build.tgz bin/updater
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: updater-build
+          path: updater-build.tgz
+
+  pre-build-scanner:
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/stackrox-io/apollo-ci:scanner-test-0.3.61
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - name: Cache Go dependencies
+        uses: ./.github/actions/cache-go-dependencies
+
+      - name: Build Scanner
+        run: make scanner-build-nodeps
+
+      - name: Bundle the build to preserve permissions
+        run: tar -cvzf scanner-build.tgz image/scanner/bin/scanner
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: scanner-build
+          path: scanner-build.tgz
+
+  style-check:
+    env:
+      ARTIFACT_DIR: /artifacts
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/stackrox-io/apollo-ci:scanner-test-0.3.61
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - name: Cache Go dependencies
+        uses: ./.github/actions/cache-go-dependencies
+
+      - name: Create artifacts dir
+        run: mkdir -p "$ARTIFACT_DIR"
```

Comment:
I took a deeper look at this `ARTIFACT_DIR`. I think you can remove it from this PR. It is only used in `store_test_results` and that returns for non openshift environments without doing anything. 

The only reason to save those Junit test results would be to display them with something like https://github.com/test-summary/action which could be done in a follow on PR if warranted (I don't think there is enough ongoing dev in stackrox/scanner to warrant it IMHO).

Reply:
I changed the implementation of `store_test_results` in this branch, but yeah, that's actually why I had comment at https://github.com/stackrox/scanner/pull/1347/files#diff-d0777657fa3fd81d23aaf7273e58aee453b04e67882517900c56daeef9b3e4c1R370. Would it make sense to remove all the usage of `ARTIFACT_DIR` _and_ all the jobs that only call `store_test_results`?

Reply:
Ah I see. I missed that. I'd say remove `store_test_results()` from any test/job that is moved to GHA. The jobs that remain in OSCI will still benefit from it.